src/Microsoft.Azure.WebJobs.Host/Blobs/Bindings/BlobAttributeBindingProvider.cs

```diff
@@ -6,181 +6,319 @@
 using System.IO;
 using System.Linq;
 using System.Reflection;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.WebJobs.Host.Bindings;
-using Microsoft.Azure.WebJobs.Host.Converters;
+using Microsoft.Azure.WebJobs.Host.Blobs.Listeners;
+using Microsoft.Azure.WebJobs.Host.Config;
 using Microsoft.Azure.WebJobs.Host.Executors;
 using Microsoft.Azure.WebJobs.Host.Indexers;
+using Microsoft.Azure.WebJobs.Host.Protocols;
 using Microsoft.Azure.WebJobs.Host.Storage;
 using Microsoft.Azure.WebJobs.Host.Storage.Blob;
 using Microsoft.WindowsAzure.Storage.Blob;
 
 namespace Microsoft.Azure.WebJobs.Host.Blobs.Bindings
 {
-    internal class BlobAttributeBindingProvider : IBindingProvider, IBindingRuleProvider
+    internal class BlobExtension : IExtensionConfigProvider,
```

Comment:
>BlobExtension [](start = 19, length = 13)

This is making Blob a true extension, using IExtensionConfigProvider.  It calls the new BindToStream rule. 
This is similar to what we did for Table and Queue.

Reply:
All the other extensions that implement IExtensionConfigProvider have names follwing the pattern "FooExtensionConfig", or "FooConfig". Also, update the file name to match class name